cs/src/Management/TunnelManagementClient.cs

```diff
@@ -1151,6 +1212,51 @@ public void Dispose()
             CancellationToken cancellation)
         {
             Requires.NotNull(tunnelPort, nameof(tunnelPort));
+            options ??= new TunnelRequestOptions();
+            options.AdditionalHeaders ??= new List<KeyValuePair<string,string>>();
+            options.AdditionalHeaders = options.AdditionalHeaders.Append(new KeyValuePair<string, string>("If-Match", "*"));
+
+            if (tunnelPort.ClusterId != null && tunnel.ClusterId != null &&
+                tunnelPort.ClusterId != tunnel.ClusterId)
+            {
+                throw new ArgumentException(
+                    "Tunnel port cluster ID is not consistent.", nameof(tunnelPort));
+            }
+
+            var portNumber = tunnelPort.PortNumber;
+            var path = $"{PortsApiSubPath}/{portNumber}";
+            var result = (await this.SendTunnelRequestAsync<TunnelPort, TunnelPort>(
+                HttpMethod.Put,
+                tunnel,
+                ManagePortsAccessTokenScopes,
+                path,
+                query: GetApiQuery(),
+                options,
+                ConvertTunnelPortForRequest(tunnel, tunnelPort),
+                cancellation))!;
+            PreserveAccessTokens(tunnelPort, result);
+
+            if (tunnel.Ports != null)
```

Comment:
If `tunnel.Ports` is null, would we want to initialize it to an empty array, then add the new port(s) instead of skipping?

Reply:
I think the Go implementation will add the ports if the property is initially null, but the java and JS implementation follow the C# pattern.

Reply:
Good catch, I made the update